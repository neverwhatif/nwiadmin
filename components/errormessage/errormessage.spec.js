import React from 'react';
import { shallow } from 'enzyme';

import ErrorMessage from './index';

describe('ErrorMessage', () => {
    const component = shallow(<ErrorMessage />);
    const componentWithTitle = shallow(<ErrorMessage title="Lorem" />);
    const componentWithDescription = shallow(<ErrorMessage description="Lorem" />);
    const componentWithDetails = shallow(<ErrorMessage details="Lorem" />);

    it('should render', () => {
        expect(component).toMatchSnapshot();
        expect(componentWithTitle).toMatchSnapshot();
        expect(componentWithDescription).toMatchSnapshot();
        expect(componentWithDetails).toMatchSnapshot();
    });
    it('should render default title', () => {
        expect(shallow(<ErrorMessage />).find('h1').text()).toBe('Error');
    });
    it('should render a title', () => {
        expect(componentWithTitle.find('h1').length).toBe(1);
        expect(componentWithTitle.find('h1').text()).toBe('Lorem');
    });
    it('should render a description', () => {
        expect(componentWithDescription.find('p.description').length).toBe(1);
        expect(componentWithDescription.find('p.description').text()).toBe('Lorem');
    });
    it('should render details', () => {
        expect(componentWithDetails.find('small').length).toBe(1);
        expect(componentWithDetails.find('small').text()).toBe('Details: Lorem');
    });
    it('should render PrimaryButton', () => {
        expect(component.find('PrimaryButton').length).toBe(1);
    });

    it('should handle reload', () => {
        window.location.reload = jest.fn();

        const primaryButton = component.find('PrimaryButton');
        primaryButton.simulate('click');

        expect(window.location.reload).toHaveBeenCalled();
    });
});
