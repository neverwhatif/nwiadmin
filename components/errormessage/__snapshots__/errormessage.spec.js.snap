// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorMessage
    description="Sorry, something appears to have gone wrong. Please contact the site administrator."
    details={null}
    title="Error"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="title"
        >
          Error
        </h1>,
        <p
          className="description"
        >
          Sorry, something appears to have gone wrong. Please contact the site administrator.
        </p>,
        null,
        <p
          className="cta"
        >
          <PrimaryButton
            onClick={[Function]}
          >
            Reload
          </PrimaryButton>
        </p>,
      ],
      "className": "root",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Error",
          "className": "title",
        },
        "ref": null,
        "rendered": "Error",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sorry, something appears to have gone wrong. Please contact the site administrator.",
          "className": "description",
        },
        "ref": null,
        "rendered": "Sorry, something appears to have gone wrong. Please contact the site administrator.",
        "type": "p",
      },
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <PrimaryButton
            onClick={[Function]}
          >
            Reload
          </PrimaryButton>,
          "className": "cta",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Reload",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Reload",
          "type": [Function],
        },
        "type": "p",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="title"
          >
            Error
          </h1>,
          <p
            className="description"
          >
            Sorry, something appears to have gone wrong. Please contact the site administrator.
          </p>,
          null,
          <p
            className="cta"
          >
            <PrimaryButton
              onClick={[Function]}
            >
              Reload
            </PrimaryButton>
          </p>,
        ],
        "className": "root",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Error",
            "className": "title",
          },
          "ref": null,
          "rendered": "Error",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sorry, something appears to have gone wrong. Please contact the site administrator.",
            "className": "description",
          },
          "ref": null,
          "rendered": "Sorry, something appears to have gone wrong. Please contact the site administrator.",
          "type": "p",
        },
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <PrimaryButton
              onClick={[Function]}
            >
              Reload
            </PrimaryButton>,
            "className": "cta",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Reload",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Reload",
            "type": [Function],
          },
          "type": "p",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ErrorMessage should render 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorMessage
    description="Sorry, something appears to have gone wrong. Please contact the site administrator."
    details={null}
    title="Lorem"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="title"
        >
          Lorem
        </h1>,
        <p
          className="description"
        >
          Sorry, something appears to have gone wrong. Please contact the site administrator.
        </p>,
        null,
        <p
          className="cta"
        >
          <PrimaryButton
            onClick={[Function]}
          >
            Reload
          </PrimaryButton>
        </p>,
      ],
      "className": "root",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lorem",
          "className": "title",
        },
        "ref": null,
        "rendered": "Lorem",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sorry, something appears to have gone wrong. Please contact the site administrator.",
          "className": "description",
        },
        "ref": null,
        "rendered": "Sorry, something appears to have gone wrong. Please contact the site administrator.",
        "type": "p",
      },
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <PrimaryButton
            onClick={[Function]}
          >
            Reload
          </PrimaryButton>,
          "className": "cta",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Reload",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Reload",
          "type": [Function],
        },
        "type": "p",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="title"
          >
            Lorem
          </h1>,
          <p
            className="description"
          >
            Sorry, something appears to have gone wrong. Please contact the site administrator.
          </p>,
          null,
          <p
            className="cta"
          >
            <PrimaryButton
              onClick={[Function]}
            >
              Reload
            </PrimaryButton>
          </p>,
        ],
        "className": "root",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lorem",
            "className": "title",
          },
          "ref": null,
          "rendered": "Lorem",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sorry, something appears to have gone wrong. Please contact the site administrator.",
            "className": "description",
          },
          "ref": null,
          "rendered": "Sorry, something appears to have gone wrong. Please contact the site administrator.",
          "type": "p",
        },
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <PrimaryButton
              onClick={[Function]}
            >
              Reload
            </PrimaryButton>,
            "className": "cta",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Reload",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Reload",
            "type": [Function],
          },
          "type": "p",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ErrorMessage should render 3`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorMessage
    description="Lorem"
    details={null}
    title="Error"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="title"
        >
          Error
        </h1>,
        <p
          className="description"
        >
          Lorem
        </p>,
        null,
        <p
          className="cta"
        >
          <PrimaryButton
            onClick={[Function]}
          >
            Reload
          </PrimaryButton>
        </p>,
      ],
      "className": "root",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Error",
          "className": "title",
        },
        "ref": null,
        "rendered": "Error",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lorem",
          "className": "description",
        },
        "ref": null,
        "rendered": "Lorem",
        "type": "p",
      },
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <PrimaryButton
            onClick={[Function]}
          >
            Reload
          </PrimaryButton>,
          "className": "cta",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Reload",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Reload",
          "type": [Function],
        },
        "type": "p",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="title"
          >
            Error
          </h1>,
          <p
            className="description"
          >
            Lorem
          </p>,
          null,
          <p
            className="cta"
          >
            <PrimaryButton
              onClick={[Function]}
            >
              Reload
            </PrimaryButton>
          </p>,
        ],
        "className": "root",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Error",
            "className": "title",
          },
          "ref": null,
          "rendered": "Error",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lorem",
            "className": "description",
          },
          "ref": null,
          "rendered": "Lorem",
          "type": "p",
        },
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <PrimaryButton
              onClick={[Function]}
            >
              Reload
            </PrimaryButton>,
            "className": "cta",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Reload",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Reload",
            "type": [Function],
          },
          "type": "p",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ErrorMessage should render 4`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorMessage
    description="Sorry, something appears to have gone wrong. Please contact the site administrator."
    details="Lorem"
    title="Error"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="title"
        >
          Error
        </h1>,
        <p
          className="description"
        >
          Sorry, something appears to have gone wrong. Please contact the site administrator.
        </p>,
        <small
          className="details"
        >
          <strong>
            Details:
          </strong>
           
          Lorem
        </small>,
        <p
          className="cta"
        >
          <PrimaryButton
            onClick={[Function]}
          >
            Reload
          </PrimaryButton>
        </p>,
      ],
      "className": "root",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Error",
          "className": "title",
        },
        "ref": null,
        "rendered": "Error",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sorry, something appears to have gone wrong. Please contact the site administrator.",
          "className": "description",
        },
        "ref": null,
        "rendered": "Sorry, something appears to have gone wrong. Please contact the site administrator.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <strong>
              Details:
            </strong>,
            " ",
            "Lorem",
          ],
          "className": "details",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Details:",
            },
            "ref": null,
            "rendered": "Details:",
            "type": "strong",
          },
          " ",
          "Lorem",
        ],
        "type": "small",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <PrimaryButton
            onClick={[Function]}
          >
            Reload
          </PrimaryButton>,
          "className": "cta",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Reload",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Reload",
          "type": [Function],
        },
        "type": "p",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="title"
          >
            Error
          </h1>,
          <p
            className="description"
          >
            Sorry, something appears to have gone wrong. Please contact the site administrator.
          </p>,
          <small
            className="details"
          >
            <strong>
              Details:
            </strong>
             
            Lorem
          </small>,
          <p
            className="cta"
          >
            <PrimaryButton
              onClick={[Function]}
            >
              Reload
            </PrimaryButton>
          </p>,
        ],
        "className": "root",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Error",
            "className": "title",
          },
          "ref": null,
          "rendered": "Error",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sorry, something appears to have gone wrong. Please contact the site administrator.",
            "className": "description",
          },
          "ref": null,
          "rendered": "Sorry, something appears to have gone wrong. Please contact the site administrator.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <strong>
                Details:
              </strong>,
              " ",
              "Lorem",
            ],
            "className": "details",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Details:",
              },
              "ref": null,
              "rendered": "Details:",
              "type": "strong",
            },
            " ",
            "Lorem",
          ],
          "type": "small",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <PrimaryButton
              onClick={[Function]}
            >
              Reload
            </PrimaryButton>,
            "className": "cta",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Reload",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Reload",
            "type": [Function],
          },
          "type": "p",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
